app/models/experimentation/notifications_experiment.rb

```diff
@@ -190,13 +203,13 @@ def notification_result(notification_id)
          successful_communication_type: successful_delivery.communication_type,
          successful_communication_created_at: successful_delivery.created_at.to_s,
          successful_delivery_status: successful_delivery.result}
-      elsif communications.exists?
+      elsif communications.exists? && !queued_delivery.present?
```

Comment:
Can we extract a `Notification#current_status` (or other name) method that returns one of `successful`, `pending`, or `failed`? We can do the experiment related things here, but the status querying seems experiment agnostic. 

Plus, we should test for this edge case.